sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2747,6 +2747,17 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * Collect a Dataset as ArrowPayload byte arrays and serve to PySpark.
+   */
+  private[sql] def collectAsArrowToPython(): Int = {
+    val payloadRdd = toArrowPayloadBytes()
+    val payloadByteArrays = payloadRdd.collect()
```

Comment:
The conversion going on in `table.to_pandas()` is working on an already loaded table, but the Arrow Readers can read multiple batches of data and output a single table.  The issue is that pyspark serializers expects the data to be "framed" with the length so I can not send that directly to the Arrow Reader.  Even with `toLocalIteratorAndServer` I would have to read each batch of data on the driver, then combine.  It would be possible to write the "framed" stream another stream without the lengths, where it can then be then be read into a single table - but I'm not sure if that added complexity is worth it.